app/controllers/mixins/ems_common_angular.rb

```diff
@@ -245,7 +245,7 @@ def set_ems_record_vars(ems, mode = nil)
 
       if ems.kind_of?(ManageIQ::Providers::Azure::CloudManager)
         ems.azure_tenant_id = params[:azure_tenant_id]
-        ems.subscription    = params[:subscription]
+        ems.subscription    = params[:subscription].blank? ? nil : params[:subscription]
```

Comment:
@blomquisg I've had this discussion with Bronagh. We don't do any sort of validation in the armrest gem for accessors. How many gems do you know check their accessors for blank values? Not many.

IMO this is a Rails/PostgreSQL issue, as I cannot think of any web app I have ever worked on where I wanted a blank form value stored as an empty string in the database. Note that the behavior is also vendor dependent. A blank string is treated as NULL in Oracle, for example. Which is a violation of the SQL standard, I realize, but works the way most people want it to work in practice IMO, at least for web apps.

I'd rather not make changes in the armrest gem. Instead, I'd rather see all params auto-stripped and converted to nil if they're blank. By default, anyway. How many .blank? checks could we do away with if we had that behavior? A lot, I bet.

Anyway, a PR for the next release, one way or another. :)